packages/backend/src/api/resources/resourceService.ts

```diff
@@ -128,6 +128,9 @@ export const resourceService = {
         logger.info(`No resource found with id ${id}`);
         return new ServiceResponse(ResponseStatus.Success, 'No resource found', null, StatusCodes.NOT_FOUND);
       }
+      const savedResource = await resourceRepository.findSavedResourceByIdAsync(id);
+      if (savedResource) await resourceRepository.deleteSavedResourceAsync(id);
```

Comment:
why are we deleting here ?

Reply:
I am deleting the same resource in the saved section, if a user already save a resource and later on the resource is deleted, it will be deleted from the saved too